src/charts/sparkline/sparkline-chart.directive.js

```diff
@@ -0,0 +1,327 @@
+
+     $scope.addDataPoint = function () {
+       $scope.data.data.push(Math.round(Math.random() * 100));
+       $scope.data.xData.push(new Date($scope.data.xData[$scope.data.xData.length - 1].getTime() + (24 * 60 * 60 * 1000)));
+     };
+   };
+ </file>
+ </example>
+ */
+angular.module('patternfly.charts').directive('pfSparklineChart', ['c3ChartDefaults',
+  function (c3ChartDefaults) {
+    'use strict';
+    return {
+      restrict: 'A',
+      scope: {
+        config: '=',
+        chartData: '=',
+        chartHeight: '@',
+        showXAxis: '@',
+        showYAxis: '@'
+      },
+      replace: true,
+      templateUrl: 'charts/sparkline/sparkline-chart.html',
+      controller: ['$scope',
+        function ($scope) {
+
+          // Create an ID for the chart based on the chartId in the config if given
+          $scope.sparklineChartId = 'sparklineChart';
+          if ($scope.config.chartId) {
+            $scope.sparklineChartId = $scope.config.chartId + $scope.sparklineChartId;
+          }
+
+          /*
+           * Convert the config data to C3 Data
+           */
+          $scope.getSparklineData = function (chartData) {
+            return {
+              x: chartData.xData[0],
+              columns: [
+                chartData.xData,
+                chartData.data
+              ],
+              type: 'area'
+            };
+          };
+
+          $scope.sparklineTooltip = function (scope) {
+            return {
+              contents: function (d) {
+                var tip;
+
+                if (scope.config.tooltipFn) {
+                  tip = '<div id="sparkline-tooltip" class="module-triangle-bottom">' +
+                           scope.config.tooltipFn(d) +
+                        '</div>';
+                }
+                else if (scope.config.tooltipType === 'usagePerDay') {
+                  var dayUsage = Math.round(d[0].value / scope.chartData.total * 100.0);
+                  tip = '<div id="sparkline-tooltip" class="module-triangle-bottom">' +
+                    '  <table class="c3-tooltip">' +
+                    '    <tbody>' +
+                    '      <tr>' +
+                    '        <th colspan="2">' + d[0].x.toLocaleDateString() + '</th>' +
+                    '      </tr>' +
+                    '      <tr>' +
+                    '        <td class="name">' + dayUsage + '%:' + '</td>' +
+                    '        <td class="value text-nowrap">' + d[0].value + ' ' + scope.config.totalUnits + ' ' + d[0].name + '</td>' +
+                    '      </tr>' +
+                    '    </tbody>' +
+                    '  </table>' +
+                    '</div>';
+                }
+                else if (scope.config.tooltipType === 'valuePerDay') {
+                  tip = '<div id="sparkline-tooltip" class="module-triangle-bottom">' +
+                    '  <table class="c3-tooltip">' +
+                    '    <tbody>' +
+                    '      <tr>' +
```

Comment:
Been trying to use a directive here but to no avail. I will, however, create a utility method to put the common html in one place.